packages/react/src/executors/module-federation-dev-server/module-federation-dev-server.impl.ts

```diff
@@ -53,13 +53,20 @@ function getModuleFederationConfig(
 
   let moduleFederationConfigPath = moduleFederationConfigPathJS;
 
+  // create a no-op so this can be called with issue
+  let tsNodeService;
   if (existsSync(moduleFederationConfigPathTS)) {
-    tsNodeRegister(moduleFederationConfigPathTS, tsconfigPath);
+    tsNodeService = tsNodeRegister(moduleFederationConfigPathTS, tsconfigPath, {
+      transpileOnly: true,
+    });
     moduleFederationConfigPath = moduleFederationConfigPathTS;
   }
 
   try {
     const config = require(moduleFederationConfigPath);
+    if (tsNodeService) {
```

Comment:
Should we call `registerTsProject` instead from `@nx/js` plugin, which as a clean-up function that calls `.enabled(false)` for `ts-node`?

Reply:
I actually think `getTranspiler` might be better from `nx`